docs/preview/yamlgettingstarted-triggers.md

```diff
@@ -0,0 +1,96 @@
+# YAML getting started - YAML triggers (not yet available, for discussion only)
+
+## Simple CI trigger syntax
+
+On push, the YAML file from the branch that was updated, is used to determine whether
+to queue a CI build.
+
+For example, if master is updated, the trigger defined in master will be used. If a different
+branch is updated, the YAML file from that branch will be used to determine whether to queue a
+build.
+
+```yaml
+trigger:
+- master
+- releases/*
+```
+
+## CI is opt-out
+
+When not specified, the default trigger for build would be all branches.
+
+For build this behavior makes sense. Especially when we start supporting on-push defintion creation from any branch. And on-push resource authorization from any branch.
+
+For release, CI trigger would likely be opt-in? However with YAML, release will be tightly coupled with a repository. So opt-out CI behavior might also make sense for release? Or opt-out, but the default for RM only includes master?
+
+```yaml
+trigger: false | none # Turn off CI
+```
+
+## Full CI trigger syntax
+
+```yaml
+trigger:
+  batch: bool | number
+  branches:
+    include: []
+    exclude: []
+  paths:
+    include: []
+    exclude: []
+```
+
+## Schedules (future; needs cleanup)
+
+```yaml
+schedules:
+- schedule: name
+  whenUnchanged: bool
+  days: # Also supports a string instead of an array: "<DAY>" or "All" or "Weekdays"
+  - sun
+  - mon
+  - tues
+  - wed
+  - thurs
+  - fri
+  - sat
+  timeZone: EST # Defaults to account time zone? Do we know that info? We might need to maintain a mapping for this abbrev, it is not in system.timeZoneInfo.
+  times:
+  - 03:00 am
+  - 03:00 pm
+  branches:
+    include: []
```

Comment:
What are the branch rules here?  What happens if the branches are empty?  What happens if I specify 'master' in master and 'master, release/1.0' in the my release branch.  What happens if I specify release/1.0 in my dev branch?

Reply:
In this case we have been discussing if we should treat schedules a bit differently than CI.  Our thought is that the schedules are read form the repos default branch instead of trying to track and merge across all branches in a repo. In that model you would need to specify all schedules for all branches in master.

Reply:
@chrisrpatterson That is a bit odd though, since you have to alter process in one branch (in the target branch for the schedule), but policy (scheduling) in another.

My thought would be that the schedule would be read from the target branch, just like CI.  Since updating the branch policy is a commit, VSTS would notice the commit, update the schedule for that branch if necessary, etc.

Reply:
Yes, however,  a branch can be a pattern like releases/* so if I have specified in master to run releases/* every day at 3 and then in one of the release branches I change releases/* to once a week who wins?

Reply:
The branch wins in that case.  What I see here is that by specifying as many branches as you want in a single definition, you're improving the mergeability of these files across branches.  However, as a general rule, I think if the **process** (what needs to be done) comes from the branch itself, then the policy (when it needs to be done) should also be in that branch.